plugins/woocommerce-admin/client/products/hooks/use-variations-order.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import { useFormContext } from '@woocommerce/components';
+import type { ProductVariation } from '@woocommerce/data';
+
+/**
+ * Internal dependencies
+ */
+
+const KEY_SEPARATOR = ':';
+
+function getVariationKey( variation: ProductVariation ) {
+	return `${ variation.id }${ KEY_SEPARATOR }${ variation.menu_order }`;
+}
+
+function getVariationId( { key }: JSX.Element ) {
+	return typeof key === 'string'
+		? Number.parseInt( key.split( KEY_SEPARATOR )[ 0 ], 10 )
+		: 0;
+}
+
+function getVariationOrder( { key }: JSX.Element ) {
+	return typeof key === 'string'
+		? Number.parseInt( key.split( KEY_SEPARATOR )[ 1 ], 10 )
+		: Number.MAX_SAFE_INTEGER;
+}
+
+function sort(
+	variations: ProductVariation[],
+	currentPage: number,
+	{ variationOrders }: ProductVariationsOrder
+) {
+	if ( ! variationOrders || ! variationOrders[ currentPage ] )
+		return variations;
+
+	const currentPageOrders = variationOrders[ currentPage ];
+
+	return [ ...variations ].sort( ( a, b ) => {
+		if ( ! currentPageOrders[ a.id ] || ! currentPageOrders[ b.id ] )
+			return 0;
+		return currentPageOrders[ a.id ] - currentPageOrders[ b.id ];
+	} );
+}
+
+export default function useVariationsOrder( {
+	variations,
+	currentPage,
+}: UseVariationsOrderInput ): UseVariationsOrderOutput {
+	const { setValue, values } = useFormContext< ProductVariationsOrder >();
+
+	function onOrderChange( items: JSX.Element[] ) {
+		const minOrder = Math.min( ...items.map( getVariationOrder ) );
+
+		setValue( 'variationOrders', {
```

Comment:
We have a few more cases of `variationOrders` that need to be updated to `variationsOrder` and `updateVariationOrders` that need to `updateVariationsOrder`.

Reply:
Ohh, nice catch man. -> [362ae82](https://github.com/woocommerce/woocommerce/pull/36109/commits/362ae822d113c07ef71b2b46f509f2a4e09a5913)